### PR TITLE
PUT and PATCH without specifying ID

### PR DESCRIPTION
**Pull request by specialunderwear**
Since construct_id_filter can build a uniquely identifying query from the

posted data, we can have update on the list view. This comes invery handy if
you are synchronizing data and have idea what the id of the objects is.

### CODE REVIEW

**Comment by specialunderwear** [comment]
kan je deze ff reviewen, is super dope feature!!!!!!!

**Comment by maerteijn** [comment]
Heb ik gedaan toch?

**Comment by specialunderwear** [comment]
nee

File: oscarapi/views/admin/product.py [hunk 1]
```diff
@@ -26,11 +27,39 @@
...
+        Tries to extract a uniquely indentifying query from the posted data
+        """
+        try:
+            automatic_filter = construct_id_filter(Product, self.request.data)
+            obj = Product.objects.get(automatic_filter)
+            self.check_object_permissions(self.request, obj)
+            return obj
+        except Product.DoesNotExist:
+            raise Http404("No %s matches the given query." % Product._meta.object_name)
+
```
[the remark this answers is not in the document]
Comment (specialunderwear) [reply oscarapi/views/admin/product.py]: No why woild you want to do that? The get_object implementation is what enables the put and post methods on this view. The super needs a pk, and the whole point of api is that we do not need the pk, we can identify the object from the data sent.

[the remark this answers is not in the document]
Comment (specialunderwear) [reply oscarapi/views/admin/product.py]: It is a super powerful api with an extremely simple and very elegant implementation.

**Comment by maerteijn** [comment]
Ok, if you fix the typo’s I’ll merge it, indeed a cool feature for the product admin api

**Pull request merged**